market-generator/src/main/java/com/market/banica/generator/service/TickGeneratorImpl.java

```diff
@@ -10,32 +9,31 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
 import java.util.Map;
 import java.util.Timer;
-import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.stream.Collectors;
 
 @Component
 public class TickGeneratorImpl implements TickGenerator {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TickGeneratorImpl.class);
 
     private static final Timer TICK_TIMER = new Timer();
```

Comment:
I think that behind this timer there is a single thread: https://docs.oracle.com/javase/7/docs/api/java/util/Timer.html - Corresponding to each Timer object is a single background thread that is used to execute all of the timer's tasks, sequentially. Here the purpose is to generate multiple ticks asynchronously but not sequentially, maybe ShceduledExecutorService is a better option in this case

Reply:
I read the docs but did a few test regarding the performance of Timer and it is significantly more lightweight. When you run 4 threads running same tick at same time, it can generate the tick more accurately. 
However, I just read a comparison and regarding RuntimeExceptions it can stop the whole generation, we can change it to ScheduledExecutorService. What would be an appropriate ThreadCount if you'd like to be updated.

Reply:
Updated in another pull request https://github.com/banica-org/banica-with-pumpkin/pull/44